parliament_challenge.py

```diff
@@ -0,0 +1,149 @@
+    return speeches_list
+
+
+def filter_member_dict(member):
+    """
+    Filtering member data for relevant keys.
+    Relevant keys are tilltalsnamn, valkrets,
+    bild_url_192, uppgift and intressent_id.
+    """
+
+    if member["personuppgift"] is None:
+        email = None
+    else:
+        for e in member["personuppgift"]["uppgift"]:
+            if e["kod"] == "Officiell e-postadress":
+                uppgift = e["uppgift"]
+                email = uppgift[0]
+
+    membersDict = {
+        "tilltalsnamn": member.get("tilltalsnamn"),
+        "valkrets": member.get("valkrets"),
+        "uppgift": email,
+        "bild_url_192": member.get("bild_url_192"),
+        "intressent_id": member.get("intressent_id")
+    }
+    return membersDict
+
+
+def get_member_data(intressent_id):
+    """
+    Get a member data from the member's api
+    and filter for relevant keys.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    members = 'personlista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?iid={}&utformat={}'.format(domain, members, intressent_id, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        # One member data being returned at a time.
+        if data["personlista"]["@hits"] == "1":
+            member = data["personlista"]["person"]
+            filtered_member = filter_member_dict(member)
+            return filtered_member, response.status_code
+        # In case a member detail is non-existent,
+        # returning an empty dict to indicate that 
+        # the detail does not exist in the member data.
+        elif data["personlista"]["@hits"] == "0":
+            nullMemberDict = {}
+            return nullMemberDict, response.status_code
+    else:
+        return {}, response.status_code
+
+
+def get_speeches_data(anftyp, size):
+    """
+    Get speeches data from speech's api
+    and filter for relevant keys.
+    For each speech we query the member api 
+    to get the member data and 
+    then merge it with the speech data.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    speeches = 'anforandelista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?anftyp={}&sz={}&utformat={}'.format(domain, speeches, anftyp, size, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        # When a client requests for more than one speech.
+        if data["anforandelista"]["@antal"] > "1":
+            speeches = data["anforandelista"]["anforande"]
+            filtered_speeches = filter_speeches_dict(speeches)
+            for speech in filtered_speeches:
+                member_data, code = get_member_data(speech["intressent_id"])        
+                speech.update(member_data)
+            return filtered_speeches, code
+        # When a client requests only for one speech.
+        # As a dict is returned, that speech is stored 
+        # in a list and then filtered.
```

Comment:
Same here, the client is not requesting for one speech! Revise the comment, doesn't make sense.

Reply:
Just indicate in very simple words what the logic is about in the case when the speech api returns only 1 speech. You know that it returns a dict and not a list, so we need to handle this situation and adapt it to the `filter_speeches_dict` which accepts a list instead of a dict, so we convert this into a list to be accepted by the `filter_speeches_dict` function.